lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php

```diff
@@ -124,7 +125,19 @@ public function exec($statement)
             throw SQLSrvException::fromSqlSrvErrors();
         }
 
-        return sqlsrv_rows_affected($stmt);
+        $affectedRowCount = sqlsrv_rows_affected($stmt);
+
+        $stmt = sqlsrv_query($this->conn, 'SELECT @@IDENTITY');
```

Comment:
This is what I was talking about before showing off this approach ("performance impact"). If you have a look at how the different PDO drivers implement `lastInsertId()` you will see that each of them actually trigger a query. `pdo_sqlsrv` uses [the exact same](https://github.com/Microsoft/msphpsql/blob/a02691e277d5804c3d8b28fcdfee31e6fd2df8ab/source/pdo_sqlsrv/pdo_dbh.cpp#L31), `pdo_pgsql` uses `SELECT LASTVAL()`, `pdo_mysql` uses `SELECT LAST_INSERT_ID()`. However I am not sure but I believe those are session related information calls which do not go through the server at all. Would need to investigate this.